### ISSUE TITLE
Freemarker Exception when using Enunciate on a Spring-Boot MVC POST method with @ResponseStatus

### ISSUE THREAD

**Issue by cheppsn** [stoicflame/enunciate#302]
I'm trying to use _enunciate 2.2_ with Gradle and a Spring-Boot project with Spring Web-MVC.

My controller contains a POST method like this:

    @RequestMapping(path = "/companies", method = RequestMethod.POST)
    @ResponseStatus(code = HttpStatus.CREATED)
    @ResponseBody
    public Company createCompany(@RequestBody Company company) throws BadRequestException {
    	...
    	return company;
    }
    
It seems that the `@ResponseStatus(code=HttpStatus.CREATED)` annotation leads to the follwing exception

    The following has evaluated to null or missing:
    ==> responseCode.condition

Full stack trace:

    chb@thinkpad MINGW64 /e/projects/java/tryout-spring-rest-docs (master)
    $ ./gradlew enunciate
    :enunciate
    warning: [options] bootstrap class path not set in conjunction with -source 1.5
    warning: [options] source value 1.5 is obsolete and will be removed in a future release
    warning: [options] target value 1.5 is obsolete and will be removed in a future release
    warning: [options] To suppress warnings about obsolete options, use -Xlint:-options.
    4 warnings
    warning: [options] bootstrap class path not set in conjunction with -source 1.5
    warning: [options] source value 1.5 is obsolete and will be removed in a future release
    warning: [options] target value 1.5 is obsolete and will be removed in a future release
    warning: [options] To suppress warnings about obsolete options, use -Xlint:-options.
    4 warnings
    Error executing FreeMarker template
    :enunciate FAILED
    
    FAILURE: Build failed with an exception.
    
    * What went wrong:
    Execution failed for task ':enunciate'.
    > com.webcohesion.enunciate.EnunciateException: freemarker.core.InvalidReferenceException: The following has evaluated to null or missing:
    ==> responseCode.condition  [in template "jar:file:/home/chb/.gradle/caches/modules-2/files-2.1/com.webcohesion.enunciate/enunciate-docs/2.2.0/f3bfdc6ce2b2be46eaa20a02d0ca7b93aab3b795/enunciate-docs-2.2.0.jar!/com/webcohesion/enunciate/modules/docs/docs.fmt" at line 604, column 19]
    
    ----
    Tip: It's the step after the last dot that caused this error, not those before it.
    ----
    Tip: If the failing expression is known to be legally refer to something that's null or missing, either specify a default value like myOptionalVar!myDefault, or use [#if myOptionalVar??]when-present[#else]when-missing[/#if]. (These only cover the last step of the expression; to cover the whole expression, use parenthesis: (myOptionalVar.foo)!myDefault, (myOptionalVar.foo)??
    ----
    
    ----
    FTL stack trace ("~" means nesting-related):
            - Failed at: ${responseCode.condition}  [in template "jar:file:/home/chb/.gradle/caches/modules-2/files-2.1/com.webcohesion.enunciate/enunciate-docs/2.2.0/f3bfdc6ce2b2be46eaa20a02d0ca7b93aab3b795/enunciate-docs-2.2.0.jar!/com/webcohesion/enunciate/modules/docs/docs.fmt" in macro "processResourceGroup" at line 604, column 17]
            ~ Reached through: #nested  [in template "jar:file:/home/chb/.gradle/caches/modules-2/files-2.1/com.webcohesion.enunciate/enunciate-docs/2.2.0/f3bfdc6ce2b2be46eaa20a02d0ca7b93aab3b795/enunciate-docs-2.2.0.jar!/com/webcohesion/enunciate/modules/docs/docs.fmt" in macro "boilerplate" at line 119, column 9]
            ~ Reached through: @boilerplate title=(title + ": " + re...  [in template "jar:file:/home/chb/.gradle/caches/modules-2/files-2.1/com.webcohesion.enunciate/enunciate-docs/2.2.0/f3bfdc6ce2b2be46eaa20a02d0ca7b93aab3b795/enunciate-docs-2.2.0.jar!/com/webcohesion/enunciate/modules/docs/docs.fmt" in macro "processResourceGroup" at line 472, column 5]
            - Reached through: @file name=(resourceGroup.slug + ".ht...  [in template "jar:file:/home/chb/.gradle/caches/modules-2/files-2.1/com.webcohesion.enunciate/enunciate-docs/2.2.0/f3bfdc6ce2b2be46eaa20a02d0ca7b93aab3b795/enunciate-docs-2.2.0.jar!/com/webcohesion/enunciate/modules/docs/docs.fmt" in macro "processResourceGroup" at line 471, column 3]
            - Reached through: @processResourceGroup resourceGroup=r...  [in template "jar:file:/home/chb/.gradle/caches/modules-2/files-2.1/com.webcohesion.enunciate/enunciate-docs/2.2.0/f3bfdc6ce2b2be46eaa20a02d0ca7b93aab3b795/enunciate-docs-2.2.0.jar!/com/webcohesion/enunciate/modules/docs/docs.fmt" at line 215, column 11]
            ~ Reached through: #nested  [in template "jar:file:/home/chb/.gradle/caches/modules-2/files-2.1/com.webcohesion.enunciate/enunciate-docs/2.2.0/f3bfdc6ce2b2be46eaa20a02d0ca7b93aab3b795/enunciate-docs-2.2.0.jar!/com/webcohesion/enunciate/modules/docs/docs.fmt" in macro "boilerplate" at line 119, column 9]
            ~ Reached through: @boilerplate pagenav=pagenav  [in template "jar:file:/home/chb/.gradle/caches/modules-2/files-2.1/com.webcohesion.enunciate/enunciate-docs/2.2.0/f3bfdc6ce2b2be46eaa20a02d0ca7b93aab3b795/enunciate-docs-2.2.0.jar!/com/webcohesion/enunciate/modules/docs/docs.fmt" at line 170, column 3]
            - Reached through: @file name=indexPageName  [in template "jar:file:/home/chb/.gradle/caches/modules-2/files-2.1/com.webcohesion.enunciate/enunciate-docs/2.2.0/f3bfdc6ce2b2be46eaa20a02d0ca7b93aab3b795/enunciate-docs-2.2.0.jar!/com/webcohesion/enunciate/modules/docs/docs.fmt" at line 159, column 1]
    ----
    
    * Try:
    Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output.
    
    BUILD FAILED
    
The error no longer occurs when I remove `@ResponseStatus`.

**Comment by stoicflame** [comment]
Thanks for the report.

**Comment by stoicflame** [comment]
Temporary workaround might be to supply a condition:

```
@ResponseStatus(
  code = HttpStatus.CREATED,
  condition = "If the request was processed successfully."
)
```

**Comment by cheppsn** [comment]
Unfortunately, `org.springframework.web.bind.annotation.ResponseStatus` does not have a `condition` field, there is only `code` and `reason`. But I can comment out the _Freemarker Template_ parts that cause the exception for now.

**Comment by charliepjive** [comment]
Hi Ryan,
it appears the issue I opened (#327) is the same as this one. You can either close it or merge it (if allowed, sorry I do not have a lot of experience on GITHUB)

Cheers,
Salvatore

**Comment by stoicflame** [comment]
Fixed at 1fbf68e.

**Comment by stoicflame** [comment]
[Enunciate 2.3 has been released](https://github.com/stoicflame/enunciate/releases/tag/v2.3.0).
